Makefile

```diff
@@ -0,0 +1,81 @@
+help:
+	@echo "check - clean the environment, install, lint, and run tests"
+	@echo "clean - remove Python file artifacts"
+	@echo "clean-all - remove *all* build artifacts"
+	@echo "clean-env - remove virtual environment"
+	@echo "deps - install the dependencies"
+	@echo "env - set up the virtual environment"
+	@echo "lint - check Python lint"
+	@echo "test - run tests"
+	@echo "test-deps - install the dependencies needed for running tests"
+
+###########
+# VARIABLES
+###########
+ENV := env/
+BIN := $(ENV)bin/
+FLAKE8 := $(BIN)flake8
+NOSETESTS := $(BIN)nosetests
+PIP := $(BIN)pip
+PYTHON := $(BIN)python
+
+#######
+# SETUP
+#######
+$(PYTHON):
+	virtualenv $(ENV)
+
+$(PIP):
+	virtualenv $(ENV)
+
+.PHONY: env
+env: $(PYTHON)
+
+##############
+# DEPENDENCIES
+##############
+.PHONY: test-deps
+test-deps: $(PIP)
+	$(PIP) install -r test-requirements.txt
+
+.PHONY: deps
+deps: $(PIP)
```

Comment:
Likewise add requirements.txt to this dependency so the deps will get regenerated when that file changes.